tensorrt_llm/llmapi/llm_args.py

```
@@ -2794,6 +2822,23 @@ def validate_attention_dp_config(self) -> 'TorchLlmArgs':
 
+    @model_validator(mode='after')
+    def validate_and_sync_sm_disagg_config(self) -> 'TorchLlmArgs':
+        """Validate SM-level disaggregation configuration."""
+        if self.sm_disagg_config is None:
+            return self
+
+        config = self.sm_disagg_config
+        if not 0 < config.context_sm_percent < 1:
+            raise ValueError(
+                "sm_disagg_config.context_sm_percent must be in the range (0, 1)"
+            )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)